backend/graphql/schema.graphql

```diff
@@ -81,6 +91,7 @@ type User {
     username: String!
     created_at: DateTime!
     updated_at: DateTime
+    roles: [Role]
```

Comment:
```suggestion
    roles: [Role!]!
```

Reply:
Roles are optional at the moment. I suspect we'll have to do more work elsewhere to make them required before I can justify designating the return of null value here as an invalid request.

Reply:
I just tried querying a user that has no role with `[Role!]!` set in the schema and observed no different to `[Role]` so now I'm not even sure what designating this as a non-nullable field is actually doing.

Reply:
We don't currently return null, we return an empty array. 

https://stackoverflow.com/questions/46770501/graphql-non-nullable-array-list/46771890